src/index.js

```diff
@@ -150,12 +150,20 @@ class IOpipeWrapperClass {
   invoke() {
     this.runHook('pre:invoke');
     try {
-      return this.userFunc.call(
+      const result = this.userFunc.call(
         this.originalIdentity,
         this.event,
         this.context,
         this.callback
       );
+      if (result && result.then && result.catch) {
+        return new Promise(() => this.callback(null, () => result))
+          .then(value => value)
+          .catch(err => {
```

Comment:
No `this.sendReport(err, () => this.originalCallback(err));` or equivalent here? If AWS treats passing an error to the original callback as an exception then that might be the simplest approach. The goal here is to not modify the original handler's behavior in any way.

Reply:
I did have a version of this that used the sendReport, though the duplication was looking wrong to me. It sounds like the best (least invasive) approach is to do the sendReport in this block and return the error.

I also haven't been able to force this Promise/then/catch block to trigger a catch, which is relevant to the comment below...

Reply:
Yeah, this makes sense the more I think about it. Throwing in the `.catch()` then relying on the try/catch to report reduces code duplication. Just not sure we want to wrap every exception as an `Error`.

Reply:
I think the cases in which this catch is invoked would be very few; keeping it in there is just defensive programming.  If this block should be hit, it'd send the report, and then throw to the enclosing context. (I'm not sure if this might mean a double report, but we do check to see if reports have been sent.)